frontend/hooks/data-fetching/analytics.ts

```diff
@@ -72,3 +78,32 @@ export const useHeatmapData = (
 
     return { data, error, isValidating };
 };
+
+export const useStatistic = (
+    courseId: number,
+    queueId: number,
+    type: Metric,
+    range: number
+) => {
+    let dateRange = "";
+    if (range) {
+        let date = new Date();
+        date.setDate(date.getDate() - range);
+        date = new Date(date.getTime() - date.getTimezoneOffset() * 60 * 1000);
+        const formattedDate = date.toISOString().split("T")[0];
+        dateRange = range ? `&date__gte=${formattedDate}` : "";
```

Comment:
Good catch. I initially implemented range as an optional parameter, allowing undefined (no range would give all entries, rather than all those >= a date). Not sure why I changed that, but do you think that I should add that back, or perhaps reserve 0 as that, or reimplement as a enum?

Reply:
Range can only be 7, 30, and 90 then? In that case you could use an enum. No need to reserve special range values like 0 or undefined since we're not using them in any components yet.

Reply:
Should I not make the hook more general though? in case we do make a component that requires a different range from that. at the very least i coiuld not deal with 0 or undefined and just require a positive range to simplify things but I'm not sure what the benefit of using an enum would be.